src/ui/Folding/Folding.ts

```diff
@@ -414,12 +417,35 @@ export class Folding extends Component {
     const queryResults = data.results;
 
     const getResult: (result: IQueryResult) => IQueryResult = this.options.getResult || Folding.defaultGetResult;
-    queryResults.results = _.map(queryResults.results, getResult);
+    queryResults.results = map(queryResults.results, getResult);
+
+    if (this.options.rearrange) {
+      queryResults.results.forEach(result => {
+        result.childResults = sortBy(result.childResults, result => Utils.getFieldValue(result, this.options.rearrange.sort));
+        if (this.shouldBeReversed(result.childResults)) {
+          result.childResults = result.childResults.reverse();
+        }
+      });
+    }
+
     this.addLoadMoreHandler(<IQueryResult[]>queryResults.results, data.query);
   }
 
+  private shouldBeReversed(childResults: IQueryResult[]) {
+    if (this.options.rearrange.direction != 'descending') {
```

Comment:
Could the condition check if the value `=== 'ascending'` and return `true`?

Reply:
That would not be correct, since it would reverse improperly if we want to sort on a field that is not available on child results.

I can change it to 'direction == "ascending"=> false', which is simpler to read.